### PR TITLE
Add Vartype.INTEGER

### PR DESCRIPTION
**Pull request by arcondello**
Closes https://github.com/dwavesystems/dimod/issues/664

### CODE REVIEW

File: dimod/vartypes.py [hunk 1]
```diff
@@ -75,24 +90,38 @@ class Vartype(enum.Enum):
...
+def _vartype_miss(vartype, extended):
+    if extended:
+        candidates = ("Vartype.SPIN, 'SPIN', {-1, 1}, "
+                      "Vartype.BINARY, 'BINARY', {0, 1},"
+                      "Vartype.INTEGER, or 'INTEGER'")
+    else:
+        candidates = ("Vartype.SPIN, 'SPIN', {-1, 1}, "
+                      "Vartype.BINARY, 'BINARY', or {0, 1}")
+    return TypeError(f"expected input vartype to be one of: {candidates!s}; "
+                     f"recieved {vartype!r}.")
```
Comment (randomir) [inline dimod/vartypes.py:119]: typo

File: dimod/vartypes.py [hunk 2]
```diff
@@ -75,24 +90,38 @@ class Vartype(enum.Enum):
...
+
+
+# In the future we may wish to make extended default to True. However, for
+# now even raining a PendingDeprecationWarning would be a big change.
```
Comment (randomir) [inline dimod/vartypes.py:123]: raining?

File: dimod/vartypes.py [hunk 3]
```diff
@@ -75,24 +90,38 @@ class Vartype(enum.Enum):
...
-    DISCRETE = int
+# Deprecated alias
+ExtendedVartype = Vartype
 
 
 SPIN = Vartype.SPIN
 BINARY = Vartype.BINARY
-DISCRETE = ExtendedVartype.DISCRETE
+INTEGER = Vartype.INTEGER
+DISCRETE = Vartype.DISCRETE
```
Comment (randomir) [inline dimod/vartypes.py:104]: sorting by line length for the win! :grin:

File: dimod/vartypes.py [hunk 4]
```diff
@@ -55,13 +55,28 @@
...
+class Integers(Container):
+    """Container for testing integer membership."""
+    def __contains__(self, item):
+        try:
+            return int(item) == item
+        except TypeError:
+            return False
+
+
+integers = Integers()
```
Comment (randomir) [inline dimod/vartypes.py:77]: I'm sure there's a good reason for this, but it's not obvious to me. Where is membership testing used?

**Review by randomir — APPROVED** [review]
LGTM

File: dimod/vartypes.py (hunk 4 above)
Comment (arcondello) [reply dimod/vartypes.py:77]: An example would be something like https://github.com/dwavesystems/dimod/blob/63d17243b2aab52852df8e4e8760150ee2bc40c2/dimod/core/bqm.py#L1372

File: dimod/vartypes.py (hunk 2 above)
Comment (arcondello) [reply dimod/vartypes.py:123]: :umbrella:

File: dimod/vartypes.py (hunk 4 above)
Comment (randomir) [reply dimod/vartypes.py:77]: Of course. Enum's value is actually a domain of the type. Forgot that.

**Pull request merged**